pyvisa_py/serial.py

```diff
@@ -184,20 +173,20 @@ def write(self, data: bytes) -> Tuple[int, StatusCode]:
 
         """
         logger.debug("Serial.write %r" % data)
-        end_out, _ = self.get_attribute(ResourceAttribute.asrl_end_out)
         send_end, _ = self.get_attribute(ResourceAttribute.send_end_enabled)
+        end_out, _ = self.get_attribute(ResourceAttribute.asrl_end_out)
+        data_bits, _ = self.get_attribute(constants.ResourceAttribute.asrl_data_bits)
```

Comment:
I reordered `end_out` and `send_end` so that they match `pyvisa_sim`: https://github.com/pyvisa/pyvisa-sim/blob/17323c60b9e50f8cca4b983d42dd26a6ada877e3/pyvisa_sim/sessions/serial.py#L76-L78

I opted to not rename `end_out` to `asrl_end` to keep the diff down.